_episodes/08-putting-it-all-together.md

```diff
@@ -86,209 +89,235 @@ also change the name of the columns in the DataFrame like this:
+ax1.set_xlabel("Hindfoot length")
+ax1.tick_params(labelsize=16, pad=8)
+fig.suptitle('Scatter plot of weight versus hindfoot length', fontsize=15)
 ~~~
 {: .language-python}
 
-A single figure can also include multiple plots in a grid pattern. The
-`subplot()` command specifies the number of rows, the number of columns, and
-the number of the space in the grid that particular plot is occupying:
+![Extended version of scatter plot surveys](../fig/08_scatter_surveys_extended.png)
+
+To retrieve the matplotlib figure object from plotnine for customization, use the `draw()` function in plotnine:
 
 ~~~
-    plt.figure(1)
+import plotnine as p9
+myplot = (p9.ggplot(data=surveys, 
+                    mapping=p9.aes(x='hindfoot_length', y='weight')) +
+              p9.geom_point())
+
+# convert output plotnine to a matplotlib object
+my_plt_version = myplot.draw()
+
+# Provide further adaptations with matplotlib:
+p9_ax = my_plt_version.axes[0] # each subplot is an item in a list
+p9_ax.set_xlabel("Hindfoot length")
+p9_ax.tick_params(labelsize=16, pad=8)
+p9_ax.set_title('Scatter plot of weight versus hindfoot length', fontsize=15)
+my_plt_version
+~~~
+{: .language-python}
 
-    plt.subplot(2,2,1)  # Two row, two columns, position 1
-    plt.plot(t, t, 'r--', label='linear')
+![Extended version of plotnine scatter plot](../fig/08_scatter_surveys_plotnine.png)
+
+> ## Challenge - Pandas and matplotlib
+> Load the streamgage data set with Pandas, subset the week of the 2013 Front Range flood
+> (September 9 through 15) and create a hydrograph (line plot) of the discharge data using
+> Pandas, linking it to an empty maptlotlib `ax` object. Adapt the title, x-axis and y-axis label 
+> using matplotlib.
+>
+> > ## Answers
+> >
+> > ~~~
+> > discharge = pd.read_csv("data/bouldercreek_09_2013.txt", 
+> >                         skiprows=27, delimiter="\t", 
+> >                         names=["agency", "site_id", "datetime",
+> >                                "timezone", "discharge", "discharge_cd"])
+> > discharge["datetime"] = pd.to_datetime(discharge["datetime"])
+> > front_range = discharge[(discharge["datetime"] >= "2013-09-09") & 
+> >                         (discharge["datetime"] < "2013-09-15")]
+> > 
+> > fig, ax = plt.subplots()
+> > front_range.plot(x ="datetime", y="discharge", ax=ax)
+> > ax.set_xlabel("") # no label
+> > ax.set_ylabel("Discharge, cubic feet per second")
+> > ax.set_title(" Front Range flood event 2013")
+> > ~~~
+> > {: .language-python}
+> >
+> > ![Flood event plot](../fig/08_flood_event.png)
+> {: .solution}
+{: .challenge}
 
-    plt.subplot(2,2,2)  # Two row, two columns, position 2
-    plt.plot(t, t**2, 'bs-', label='square')
+### Saving matplotlib figures
 
-    plt.subplot(2,2,3)  # Two row, two columns, position 3
-    plt.plot(t, t**3, 'g^:', label='cubic')
+Once satisfied with the resulting plot, you can save the plot with the `.savefig(*args)` command from matplotlib:
```

Comment:
Regarding command vs. method: do we want to be accurate here or do we want to sacrifice correctness for clarity?